memory/memhotplug.py

```diff
@@ -0,0 +1,250 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Abdul Haleem <abdhalee@linux.vnet.ibm.com>
+
+import os
+import glob
+import re
+import multiprocessing
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.utils.software_manager import SoftwareManager
+
+
+blocks_hotpluggable = []
+mem_path = '/sys/devices/system/memory'
+errorlog = ['WARNING: CPU:', 'Oops',
+            'Segfault', 'soft lockup',
+            'Unable to handle paging request',
+            'rcu_sched detected stalls',
+            'NMI backtrace for cpu',
+            'Call Trace:', 'WARNING: at',
+            'INFO: possible recursive locking detected',
+            'Kernel BUG at', 'Kernel panic - not syncing:',
+            'double fault:']
+
+
+def online(block):
+    try:
+        memory.hotplug(block)
+    except IOError:
+        print "memory%s : Resource is busy" % block
+        pass
+
+
+def offline(block):
+    try:
+        memory.hotunplug(block)
+    except IOError:
+        print "memory%s : Resource is busy" % block
+        pass
+
+
+def get_hotpluggable_blocks(path):
+    mem_blocks = []
+    blocks_total = glob.glob(path)
+    for mem_blk in blocks_total:
+        block = re.findall("\d+", os.path.basename(mem_blk))[0]
+        block = re.sub(r'^\s*$', '', block)
+        if memory.is_hot_pluggable(block):
+            mem_blocks.append(block)
+    return mem_blocks
+
+
+def collect_dmesg(object):
+    object.whiteboard = process.system_output("dmesg")
+
+
+class memstress(Test):
+
+    '''
+    Stress test to excersize memory component
+
+    This test performs memory hotunplug/hotplug tests with below scenarios:
+       1. hotunplug one by one in a loop for all
+       2. Toggle memory blocks by making off/on in a loop
+       3. hot unplug % of memory for different ratios
+       4. dlpar memory hotplug using  drmgr
+       5. shared resource : dlpar in CMO mode
+       6. try hotplug each different numa node memblocks
+       7. run stress memory in background
+
+    :avocado: tags=memory,power,privileged
+    '''
+
+    def setUp(self):
+
+        if not memory.check_hotplug():
+            self.cancel("UnSupported : memory hotplug not enabled\n")
+
+        if os.geteuid() != 0:
+            exit("You need to have root privileges to run this script."
+                 "\nPlease try again, using 'sudo'. Exiting.")
+        sm = SoftwareManager()
+        deps = ['stress']
+        for package in deps:
```

Comment:
here also please 
`for package in ['stress']:`

Reply:
one question is this package list is going to expand in future if yes i think we can drop this for loop and in future when we  identify new dependencies implement for loop 

it can be just like 

```
  if not sm.check_installed("stress") and not sm.install("stress"):
                    self.cancel(' stress is needed for the test to be run' ) 
```

Reply:
Yes, :+1: